### PR TITLE
support next structure in acknxt

### PR DESCRIPTION
**Pull request by ripienaar**
Signed-off-by: R.I.Pienaar <rip@devco.net>

/cc @nats-io/core

### CODE REVIEW

File: server/consumer.go [hunk 1]
```diff
@@ -1061,32 +1059,30 @@ func (o *Consumer) needAck(sseq uint64) bool {
 
 // Helper for the next message requests.
 func nextReqFromMsg(msg []byte) (time.Time, int, bool, error) {
-	req := strings.TrimSpace(string(msg))
-	if len(req) == 0 {
-		return time.Time{}, 1, false, nil
+	req := bytes.TrimSpace(msg)
```
Comment (derekcollison) [inline server/consumer.go:1062]: Would make `var req []byte` and set only once based on if.

Comment (derekcollison) [reply server/consumer.go:1062]: Might pull off prefix at upper layers and not here.

**Review by derekcollison — CHANGES_REQUESTED** [review]

**Comment by ripienaar** [comment]
updated

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**